core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -179,6 +179,63 @@ int ti_sci_get_revision(struct ti_sci_msg_resp_version *rev_info)
 	return 0;
 }
 
+/**
+ * ti_sci_device_set_state() - Set device state
+ *
+ * @id:		Device identifier
+ * @flags:	Flags to setup for the device
+ * @state:	State to move the device to
+ *
+ * Return: 0 if all goes well, else appropriate error message
+ */
+static int ti_sci_device_set_state(uint32_t id, uint32_t flags, uint8_t state)
+{
+	struct ti_sci_msg_req_set_device_state req = { };
+	struct ti_sci_msg_resp_set_device_state resp = { };
+	struct ti_sci_xfer xfer = { };
+	int ret;
+
+	ret = ti_sci_setup_xfer(TI_SCI_MSG_SET_DEVICE_STATE, flags,
+				&req, sizeof(req),
+				&resp, sizeof(resp),
+				&xfer);
+	if (ret)
+		return ret;
+
+	req.id = id;
+	req.state = state;
+
+	ret = ti_sci_do_xfer(&xfer);
+	if (ret)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * ti_sci_device_get() - Get a device managed by TISCI
+ *
+ * @id:		Device Identifier
+ *
+ * Return: 0 if all goes well, else appropriate error message
+ */
```

Comment:
It is something I had not noticed in previous TI driver... you should avoid duplicating the function documentation. The doc block should preferably be in the header file (except for static functions of course).